compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/LocalLevelServiceObjectAnalysisTask.java

```diff
@@ -0,0 +1,102 @@
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.stdlib.graphql.compiler.Utils.hasCompilationErrors;
+
+public class LocalLevelServiceObjectAnalysisTask extends ServiceAnalysisTask {
+
+    public LocalLevelServiceObjectAnalysisTask(Map<DocumentId, GraphqlModifierContext> nodeMap) {
+        super(nodeMap);
+    }
+
+    private static final String PACKAGE_NAME = "graphql";
+    private static final String SERVICE_NAME = "Service";
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        if (hasCompilationErrors(context)) {
+            return;
+        }
+        ObjectConstructorExpressionNode node = (ObjectConstructorExpressionNode) context.node();
+        if (node.parent().kind() != SyntaxKind.LOCAL_VAR_DECL) {
+            return;
+        }
+        VariableDeclarationNode parentNode = (VariableDeclarationNode) node.parent();
+        if (!isGraphQLServiceObjectDeclaration(context.semanticModel(), parentNode.typedBindingPattern())) {
+            return;
+        }
+        InterfaceEntityFinder interfaceEntityFinder = getInterfaceEntityFinder(context.semanticModel());
+        ServiceValidator serviceValidator = getServiceValidator(context, node, interfaceEntityFinder);
+        if (serviceValidator.isErrorOccurred()) {
+            return;
+        }
+        Schema schema = generateSchema(context, interfaceEntityFinder, node, null);
+        DocumentId documentId = context.documentId();
+        addToModifierContextMap(documentId, node, schema);
+    }
+
+    public boolean isGraphQLServiceObjectDeclaration(SemanticModel semanticModel,
+                                                     TypedBindingPatternNode typedBindingPatternNode) {
+        TypeDescriptorNode typeDescriptorNode = typedBindingPatternNode.typeDescriptor();
+        if (typeDescriptorNode.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+            return false;
+        }
+        return isGraphqlServiceQualifiedNameReference(semanticModel, (QualifiedNameReferenceNode) typeDescriptorNode);
+    }
+
+    private boolean isGraphqlServiceQualifiedNameReference(SemanticModel semanticModel,
+                                                           QualifiedNameReferenceNode nameReferenceNode) {
+        Optional<Symbol> symbol = semanticModel.symbol(nameReferenceNode);
+        if (symbol.isEmpty()) {
+            return false;
+        }
+        TypeReferenceTypeSymbol typeSymbol = (TypeReferenceTypeSymbol) symbol.get();
+        Optional<ModuleSymbol> moduleSymbol = typeSymbol.getModule();
+        if (moduleSymbol.isEmpty()) {
+            return false;
+        }
+        if (!isPresentAndEquals(moduleSymbol.get().getName(), PACKAGE_NAME)) {
+            return false;
+        }
+        return isPresentAndEquals(typeSymbol.getName(), SERVICE_NAME);
+    }
+
+    private boolean isPresentAndEquals(Optional<String> actualValue, String expectedValue) {
```

Comment:
A function should do one thing and one thing only. When your function has the word `and`, that's a good hint that the function does more than one thing.